packages/components/src/components/files/FileTree.tsx

```diff
@@ -285,11 +297,12 @@ export class FileTree extends PureComponent<FileTreeProps, FileTreeState> {
     };
 
     // Callback to parent with actual path of selected file
-    onFileSelect = (id: string, checked: boolean, isDirectory: boolean, node: any): void => {
+    // TODO: I think this should also return a false value in the else case
```

Comment:
I had to look over to the modeEditor usage side to figure out what this returned boolean was for, but I think I understand now. It is to let the caller decide if the file selection should proceed or halt. Is that correct? If so, we should add a comment here regarding that return value. And yes, I think the else should return false in this case.

Reply:
Yes, that's correct, I've added in an explanatory comment.
For more context, this is used when a user receives a dirty state warning on attempting to navigate to another file, and 'cancel' gets clicked. In order to interrupt navigating to and highlighting the node of the new file, I had to add a check for whether or not to do the interrupt.